patches/server/0270-Honor-EntityAgeable.ageLock.patch

```diff
@@ -5,14 +5,34 @@ Subject: [PATCH] Honor EntityAgeable.ageLock
 
 
 diff --git a/src/main/java/net/minecraft/world/entity/AgeableMob.java b/src/main/java/net/minecraft/world/entity/AgeableMob.java
-index 123b125a3576903767983c93135086ca7a8ea813..c0780a8714808498390282b7fa1da1f3aacf8e2a 100644
+index 123b125a3576903767983c93135086ca7a8ea813..d165117d62fe8a55d624966e8c4b626c0f52db39 100644
 --- a/src/main/java/net/minecraft/world/entity/AgeableMob.java
 +++ b/src/main/java/net/minecraft/world/entity/AgeableMob.java
 @@ -84,6 +84,7 @@ public abstract class AgeableMob extends PathfinderMob {
      }
  
      public void ageUp(int age, boolean overGrow) {
-+        if (ageLocked) return; // Paper - GH-1459
++        if (this.ageLocked) return; // Paper - GH-1459
          int j = this.getAge();
          int k = j;
  
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+index 93f17997ac3f36d0a72e5d4e85c7e748d615801f..0dca7b0f40af9337bdda75ba40cb064f39723bf8 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/BeehiveBlockEntity.java
+@@ -295,6 +295,7 @@ public class BeehiveBlockEntity extends BlockEntity {
+     }
+ 
+     private static void setBeeReleaseData(int ticks, Bee bee) {
++        if (!bee.ageLocked) { // Paper - respect age lock
+         int j = bee.getAge();
+ 
+         if (j < 0) {
+@@ -302,6 +303,7 @@ public class BeehiveBlockEntity extends BlockEntity {
+         } else if (j > 0) {
+             bee.setAge(Math.max(0, j - ticks));
+         }
++        } // Paper - respect age lock
+ 
+         bee.setInLoveTime(Math.max(0, bee.getInLoveTime() - ticks));
```

Comment:
from the doc for setAgeLock - `Lock the age of the animal, setting this will prevent the animal from maturing or getting ready for mating.`

should the setInLoveTime call also be guarded maybe? other than that looks good

Reply:
I'm not sure that's exactly what it means. The age lock doesn't prevent the decrementing of the `Animal`s inLove field. That happens every tick its above 0. I think that sentence is either wrong, or misleading.

Look at nms Animal#aiStep, which is where I see the `inLove` field counted, there's nothing guarding it based on the entities age lock.

Reply:
It seems to me like the javadoc implies it should be impacted by the age lock, but that would be a separate thing I guess if someone were to report that